### PR TITLE
Windows support + CI to GitHub Actions

### PR DESCRIPTION
**Pull request by fanatid**
- Windows support
- CI to GitHub Actions

### CODE REVIEW

**Comment by fanatid** [comment]
CI on GitHub: https://github.com/fanatid/llparse-test-fixture/commit/f15ad7b896f8b0c7583bce70443287400368574a/checks?check_suite_id=394381331

**Comment by fanatid** [comment]
If this will be landed, https://github.com/nodejs/llparse-test-fixture/issues/1 can be closed.

File: src/fixture.ts [hunk 1]
```diff
@@ -185,26 +185,28 @@ export class Fixture {
...
 
       if (initJS) {
         jsArgs.push(`-i ${initJS}`);
       }
 
-      fs.writeFileSync(jsOut,
-        '#!/bin/sh\n' +
-        `${JS_RUNNER} ${jsArgs.join(' ')} "$1" "$2"`);
+      const content = process.platform === 'win32'
+          ? `node ${JS_RUNNER.replace(/(\s+)/g, '\\$1')} ${jsArgs.join(' ')} "%1" "%2"`
```
Comment (indutny) [inline src/fixture.ts]: Nitpick: Could you extract the `JS_RUNNER` and `jsArgs` parts of this template to a variable outside and use it here? (After reflection I think that I might have an obsession about extracting things and putting them into a variable :joy: )

File: src/native/fixture.c [hunk 2]
```diff
@@ -4,10 +4,40 @@
...
+#if defined(_MSC_VER)
+  #if defined(_MSC_EXTENSIONS)
+    #define DELTA_EPOCH_IN_MICROSECS  11644473600000000Ui64
+  #else
+    #define DELTA_EPOCH_IN_MICROSECS  11644473600000000ULL
+  #endif
+
+  #include <windows.h>
+
+  int gettimeofday(struct timeval* tv, void* tz) {
```
Comment (indutny) [inline src/native/fixture.c]: Neat polyfill! Should it be `static`?

File: src/result.ts [hunk 3]
```diff
@@ -86,16 +89,18 @@ export class FixtureResult {
 
     await onEnd;
 
+    const stdoutText = Buffer.concat(stdout).toString();
+    const stderrText = Buffer.concat(stderr).toString();
+
     if (signal) {
-      throw new Error(`Test "${name}" killed with signal: "${signal}"`);
+      throw new Error(`Test "${name}" killed with signal: "${signal}".\nstdout: ${stdoutText}\nstderr: ${stderrText}`);
```
Comment (indutny) [inline src/result.ts]: Nitpick: 80-column limit. Sorry for not configuring it in linter.

File: src/result.ts [hunk 4]
```diff
@@ -86,16 +89,18 @@ export class FixtureResult {
...
     }
 
     if (code !== 0) {
-      throw new Error(`Test "${name}" exited with code: "${code}"`);
+      throw new Error(`Test "${name}" exited with code: "${code}".\nstdout: ${stdoutText}\nstderr: ${stderrText}`);
```
Comment (indutny) [inline src/result.ts]: Nitpick: ditto

**Review by indutny — APPROVED** [review]
LGTM with comments

File: src/native/fixture.c (hunk 2 above)
Comment (fanatid) [reply src/native/fixture.c]: Yes! Thanks that pointed on this.

**Comment by fanatid** [comment]
I moved text to variables, not sure that I did it right, please let me know if I need change something else.

\+ `lib` folder should be removed from repo, any change in `src` require update it. Other `llparse` related repositories do not have it in git index...

**Review by indutny — APPROVED** [review]
LGTM

**Pull request merged**